app/controllers/api/v1/schedules_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Api
+  module V1
+    class SchedulesController < ExternalApiController
+      # This Endpoint is not using JSON API format!
+      def create
+        return forbidden unless current_user.admin?
+
+        store = Store.find_by(id: params['store-id'])
+        timetable = params['timetable']
+        return wrong_store unless store
+        return invalid_data if timetable.blank?
+
+        timetable_errors = add_timetable(store, timetable)
+
+        if timetable_errors.any?
+          render json: {error: timetable_errors}, status: :unprocessable_entity
+        else
+          render json: {message: "Added schedule for store: #{store.id}"}, status: :created
+        end
+      end
+
+      private
+
+      def add_timetable(store, timetable)
+        errors = []
+        WeekDay.transaction do
+          WeekDay.days.keys.each do |day|
+            day_time = WeekDay.find_or_create_by(store_id: store.id, day: day)
```

Comment:
Could use `find_or_initialize_by`, and only save after the data is added.

Reply:
@simaob true. I'm going to change it.